### PR TITLE
fix: mobile display

### LINKED ISSUE #204: Current mobile display is bad
### Description

The app overflows on mobile devices, forcing the user to always scroll to have basic interactions.

### Screenshots

Screen Shot 2022-10-24 at 19 24 24
Screen Shot 2022-10-24 at 19 23 49

### Additional information

_No response_

### 🥦 Browser

Other

### 👀 Have you checked if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read and understood the rules in the [Contributing Guidelines](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md)

### Are you willing to work on this issue ?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by mathiasayivor**
# Fixes Issue

**My PR closes #204 **

# 👨‍💻 Changes proposed(What did you do ?)
* Improvements on mobile display

# ✔️ Check List (Check all the applicable boxes)

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.

##  Note to reviewers

# 📷 Screenshots

### CODE REVIEW

**Comment by Dun-sin** [comment]
Why is the changes so large? 🤔

**Comment by mathiasayivor** [comment]
> Why is the changes so large? thinking

Most of the changes are class changes and the rest are lib function modifications.

**Comment by Dun-sin** [comment]
> > Why is the changes so large? thinking
> 
> Most of the changes are class changes and the rest are lib function modifications.

Just for mobile responsiveness?

**Comment by mathiasayivor** [comment]
> > > Why is the changes so large? thinking
> > 
> > 
> > Most of the changes are class changes and the rest are lib function modifications.
> 
> Just for mobile responsiveness?

Yep.

**Comment by Dun-sin** [comment]
I don't think this request solves one thing, I'm seeing end search and some none other related things in there.

Can you break them into branches and issues, for me to easily review

**Comment by mathiasayivor** [comment]
Aight, here's a breakdown of all the changes I made:
* I added @rsuite/icons library for Anonymous.jsx menu buttons (lines 85-95, 110-125, 134) 
* Moved `isSearching` && `currentChatId` status to context in order to use it for hiding the navbar when chatting with a user:
  Before:
  Screen Shot 2022-10-26 at 06 59 27
  Screen Shot 2022-10-26 at 06 58 29

  After:
  Screen Shot 2022-10-26 at 07 00 01
  Screen Shot 2022-10-26 at 06 57 41

* Used the `createClassesFromArray` util to help add multiple classes in a readable manner (vue style)
* Updated loadingText to be properly displayed on mobile devices (BuddyMatcher.jsx line 36-51)
* The rest is just CSS class updates

File: client/src/lib/utils.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
 /**
  *
  * @param {any[]} classes
  * @returns
  */
-export function createClassesFromArray(classes) {
-    return classes.filter(Boolean).join(' ');
+export function createClassesFromArray(...classes) {
+    return uniq(flattenDeep(classes)).filter(Boolean).join(' ');
 }
```
Comment (Dun-sin) [inline client/src/lib/utils.js:10]: What does this do exactly

**Comment by mathiasayivor** [comment]
> I'm seeing end search and some none other related things in there.

@Dun-sin besides `endSearch` and the changes I've explained above, what other none related things do you see?

**Comment by Dun-sin** [comment]
> > I'm seeing end search and some none other related things in there.
> 
> @Dun-sin besides `endSearch` and the changes I've explained above, what other none related things do you see?

that's all i could notice

File: client/src/lib/utils.js (hunk 1 above)
Comment (mathiasayivor) [reply client/src/lib/utils.js:10]: For adding multiple css classes easily in a readable and more maintainable way

**Comment by mathiasayivor** [comment]
> > > I'm seeing end search and some none other related things in there.
> > 
> > 
> > @Dun-sin besides `endSearch` and the changes I've explained above, what other none related things do you see?
> 
> that's all i could notice

Aight.
But is my explanation clear enough?

**Comment by Dun-sin** [comment]
> > > > I'm seeing end search and some none other related things in there.
> > > 
> > > 
> > > @Dun-sin besides `endSearch` and the changes I've explained above, what other none related things do you see?
> > 
> > 
> > that's all i could notice
> 
> Aight. But is my explanation clear enough?

yes it's clear, thanks

**Review by Dun-sin — APPROVED** [review]

**Pull request merged**